### ISSUE TITLE
Meteor update (2.9.1 to 2.10.0) takes more than 12 hours

### ISSUE THREAD

**Issue by knoxgon** [meteor/meteor#12459]
While I'm in the process of upgrading meteor to the latest version (from 2.9.1 to 2.10.0), I needed to up it to the latest version after reading the [changelogs](https://docs.meteor.com/changelog.html) yesterday.

Usually the updating process takes, in the worst case, around 2 hours which had been continuening for more than half a calendar day which is until today.

The only odd behavour that I spot is the package `montiapm:agent@2.47.2` that always keeps coming up. Note that it's not stuck in it. It jumps to another package to extract or update and jumps back working with that specific package.

This morning, I cancelled the process, removed .local folder, node_modules, restarted the computer and restarted the process. Now, it has been more than 3 hours and the same problem has been showing up again. Stuck in montiapm:agent@2.47.2.

![image](https://user-images.githubusercontent.com/24989987/213653547-ddb78353-7e49-4ff4-82cf-713a8904e13c.png)

What else is there to try?

**Comment by StorytellerCZ** [comment]
That is strange. For me all the updates have always been a matter of minutes if not seconds. The longest time usually being taken by the download of the new packages. What OS are you running on?
Also you might want to try to re-install Meteor on your system as a whole (by removing the system folder and triggering the Meteor install). It is going to take a while to download all the packages again, but maybe the cleaning of that might work.

**Comment by KoenLav** [comment]
I've experienced Meteor getting stuck while updating on Windows tons of times (more often than not) at the extracting stage (of no specific package).

What always fixed it for me: stop the update/Meteor, restart Meteor, now a Windows permission screen will show, accept, update works.

**Comment by zodern** [comment]
@KoenLav have you seen this since Meteor 2.4? Meteor used to sometimes get stuck on Windows due to the native file watching library we used, but that should be fixed now.

**Comment by KoenLav** [comment]
@zodern yes, I'm quite certain I've experienced this running 2.7 or 2.8 still. It's not hugely annoying, but does seems to happen pretty consistently.

**Comment by nachocodoner** [comment]
This issue seems to be outdated. Meteor 3 delivered many improvements to the bundler and CLI, not only through the Rspack integration. We also optimized the Meteor bundler and several internal processes, including Meteor local catalog management, which may have affected this behavior.

I encourage you to update to the latest Meteor 3 version and properly migrate your project to the [modern build stack](https://docs.meteor.com/about/modern-build-stack.html). While Windows has historically had lower performance, the latest improvements have noticeably improved the experience on my end, and performance metrics were also collected in Windows environments.

As a side note, I develop on Windows using WSL when I am not testing issues directly in a Windows environment. I also recommend using WSL, as it usually provides significant performance improvements.

I'm going to close this issue in favor of receiving new reports based on the current state of Meteor 3.x (Meteor 2.x reached end-of-life). Please share your experience, and if you still face performance problems, let us know with a proper reproduction.
